server/src/main/java/org/opensearch/cluster/coordination/LeaderChecker.java

```diff
@@ -359,6 +358,7 @@ void leaderFailed(Exception e) {
                     @Override
                     public void run() {
                         onLeaderFailure.accept(e);
+                        clusterManagerMetrics.incrementCounter(clusterManagerMetrics.leaderCheckFailureCounter, 1.0);
```

Comment:
Should the metric be emitted at line#356 irrespective of `isClosed` ?

Reply:
makes sense ; in that case lets move it to line#357 inside the `if` block

Reply:
Updated in the revision.